edith-step/src/components/TopNavbar.js

```diff
@@ -1,25 +1,100 @@
 import React from 'react';
 
-import Navbar from 'react-bootstrap/Navbar';
+import Dropdown from 'react-bootstrap/Dropdown';
 import Nav from 'react-bootstrap/Nav';
+import Navbar from 'react-bootstrap/Navbar';
 
 /**
  * Top navigation bar for user interaction with log-in, log-out
  * and other app features.
  */
 class TopNavbar extends React.Component {
+  /**
+   * @constructor
+   * @param {Object}  props for React component.
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null,
+    };
+  }
+
+  /**
+   * After the component did mount, call login servlet to get the user status.
+   */
+  componentDidMount() {
+    this.login();
+  }
+
+  /**
+   * Checks the user log-in status and sets
+   * user email in sessionstorage.
+   */
+  login() {
+    fetch('/login')
+        .then((response) => response.json())
+        .then((userInfo) => {
+          sessionStorage.setItem('logged-in', userInfo.email);
+          this.setState({user: userInfo});
```

Comment:
How is logged-in sessionStorage value used?

It seems like there is a disparity between the local state and the session storage.

Reply:
That has not been used at all. I just set that for future usage if any feature requires it in the future. But it might not be useful as react state can be utilized.

Reply:
I suggest removing it if its not needed. See https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it but take it as a reminder rather than a strict policy. You shouldn't add code because you think it will be useful, but you should add code that you intend to use.

Reply:
Thanks for the reference. Changes made.